include/aspect/postprocess/visualization/melt.h

```diff
@@ -0,0 +1,144 @@
+/*
+  Copyright (C) 2015 - 2016 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__postprocess_visualization_melt_h
+#define __aspect__postprocess_visualization_melt_h
+
+#include <aspect/postprocess/visualization.h>
+#include <aspect/simulator_access.h>
+
+#include <deal.II/numerics/data_out.h>
+
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      /**
+       * A class derived from DataPostprocessor that takes an output vector
+       * and computes a variable that represents the melt density at every point.
+       *
+       * The member functions are all implementations of those declared in the
+       * base class. See there for their meaning.
```

Comment:
I am somewhat torn about the class structure of these postprocessors. On the one hand it is nice that they are grouped in one file, because they make only sense in melt models, and they belong together. On the other hand they continue this bad design of single postprocessors that evaluate the material model and therefore *all* material properties to only output *one* of them. That was the original reason why I wrote the `MaterialProperties` postprocessor that can output *all* material properties with a single evaluation of the material model, and I would like to ask you to at least consider changing these postprocessors into the same structure. Not the smallest advantage is that it saves time during compilation to have less classes. I am still dreaming of at some point deprecating all of these "Density","Thermal expansivity", "Specific heat", .... postprocessors.